src/ripple/app/tests/Flow_test.cpp

```diff
@@ -512,6 +512,83 @@ struct Flow_test : public beast::unit_test::suite
         }
     }
 
+    void testLineQuality ()
+    {
+        testcase ("Line Quality");
+
+        using namespace jtx;
+        auto const alice = Account ("alice");
+        auto const bob = Account ("bob");
+        auto const carol = Account ("carol");
+        auto const dan = Account ("dan");
+        auto const USDA = alice["USD"];
+        auto const USDB = bob["USD"];
+        auto const USDC = carol["USD"];
+        auto const USDD = dan["USD"];
+
+        //   Dan -> Bob -> Alice -> Carol; vary bobDanQIn and bobAliceQOut
+        for (auto bobDanQIn : {80, 100, 120})
+            for (auto bobAliceQOut : {80, 100, 120})
+                for (auto&& f : {feature ("nullFeature"), featureFlowV2})
```

Comment:
Done. I wanted to write `for(f : {feature ("nullFeature"), featureFlowV2})` but the committee rejected that.